pallets/moonbeam-orbiters/src/lib.rs

```diff
@@ -0,0 +1,408 @@
+		CollatorPoolTooLarge,
+		/// The minimum deposit required to register as an orbiter has not yet been included in the
+		/// onchain storage
+		MinOrbiterDepositNotSet,
+		/// This orbiter is already associated with this collator.
+		OrbiterAlreadyInPool,
+		/// Orbiter cant leave this round
+		OrbiterCantLeaveThisRound,
+		/// This orbiter has not made a deposit
+		OrbiterDepositNotFound,
+		/// This orbiter is not found
+		OrbiterNotFound,
+	}
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(crate) fn deposit_event)]
+	pub enum Event<T: Config> {
+		/// Paid the orbiter account the balance as liquid rewards.
+		OrbiterRewarded {
+			account: T::AccountId,
+			rewards: BalanceOf<T>,
+		},
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Add an orbiter in a collator pool
+		#[pallet::weight(500_000_000)]
+		pub fn collator_add_orbiter(
+			origin: OriginFor<T>,
+			orbiter: <T::Lookup as StaticLookup>::Source,
+		) -> DispatchResult {
+			let collator = ensure_signed(origin)?;
+			let orbiter = T::Lookup::lookup(orbiter)?;
+
+			let mut collator_pool =
+				CollatorsPool::<T>::get(&collator).ok_or(Error::<T>::CollatorNotFound)?;
+			let orbiters = collator_pool.get_orbiters();
+			ensure!(
+				(orbiters.len() as u32) < T::MaxPoolSize::get(),
+				Error::<T>::CollatorPoolTooLarge
+			);
+			for orbiter_ in orbiters {
+				if orbiter_ == &orbiter {
+					return Err(Error::<T>::OrbiterAlreadyInPool.into());
+				}
+			}
+
+			// Make sure the orbiter has made a deposit. It can be an old orbiter whose deposit
+			// is lower than the current minimum (if the minimum was lower in the past), so we just
+			// have to check that a deposit exists (which means checking that the deposit amount
+			// is not zero).
+			let orbiter_deposit =
+				T::Currency::reserved_balance_named(&T::OrbiterReserveIdentifier::get(), &orbiter);
+			ensure!(
+				orbiter_deposit > BalanceOf::<T>::zero(),
+				Error::<T>::OrbiterDepositNotFound
+			);
+
+			collator_pool.add_orbiter(orbiter);
+			CollatorsPool::<T>::insert(collator, collator_pool);
+
+			Ok(())
+		}
+
+		/// Remove an orbiter from a collator pool
+		#[pallet::weight(500_000_000)]
+		pub fn collator_remove_orbiter(
+			origin: OriginFor<T>,
+			orbiter: <T::Lookup as StaticLookup>::Source,
+		) -> DispatchResult {
+			let collator = ensure_signed(origin)?;
+			let orbiter = T::Lookup::lookup(orbiter)?;
+
+			let mut collator_pool =
+				CollatorsPool::<T>::get(&collator).ok_or(Error::<T>::CollatorNotFound)?;
+
+			if !collator_pool.remove_orbiter(&orbiter) {
+				Err(Error::<T>::OrbiterNotFound.into())
+			} else {
+				Ok(())
+			}
+		}
+
+		/// Registering as an orbiter
+		#[pallet::weight(500_000_000)]
+		pub fn orbiter_register(origin: OriginFor<T>) -> DispatchResult {
+			let orbiter = ensure_signed(origin)?;
+
+			let min_orbiter_deposit = MinOrbiterDeposit::<T>::get();
+			ensure!(
+				min_orbiter_deposit > Zero::zero(),
+				Error::<T>::MinOrbiterDepositNotSet
+			);
```

Comment:
I think they will, because this is probably set in the genesis or through an extrinisc. If the origin that is allowed to register orbiters in SignedOrigin, then it can call this extrinsic before we set the minimum deposit. One thing we can do is update the value in the runtime upgrade, or even use a non-zero default value for this storage

Reply:
I finally decided on a migration to initialize the storage value `MinOrbiterDeposit`, and I'm going to add in the documentation that you have to do this migration when adding this pallet, so I think we can remove this control.

Reply:
Sounds good

Reply:
Finally, we handle this case with an OptionQuery